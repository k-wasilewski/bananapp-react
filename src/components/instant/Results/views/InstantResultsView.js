import React, {Component} from 'react';
import {Link} from 'react-router-dom';

class InstantResultsView extends Component {
    render() {
        const content = (this.props.img===undefined && this.props.days===undefined
            && this.props.accuracy===undefined) ?
            (<h3>File type not supported or it's not a banana, try again</h3>)
            :
            (<React.Fragment>
                <img src={this.props.img} className='img' alt='your banana'/>
                <h3> Your banana is { this.props.days } old, </h3>
                <h3> with { Number((this.props.accuracy*100).toFixed(2)) }% certainty</h3>
            </React.Fragment>
                );

        return (
            <div className='App'>
                <header className='App-header'>
                    {content}
                    <Link to='/'>
                        <button variant='outlined'>
                            Back
                        </button>
                    </Link>
                </header>
            </div>
        );
    }
}
export default InstantResultsView;