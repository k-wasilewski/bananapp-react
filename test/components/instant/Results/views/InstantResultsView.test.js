import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../../../src/redux/store";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import InstantResultsView from "../../../../../src/components/instant/Results/views/InstantResultsView";

describe("InstantResultsView rendering specification", () => {
    it('InstantResultsView is rendered', () => {
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <InstantResultsView />
                </BrowserRouter>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});