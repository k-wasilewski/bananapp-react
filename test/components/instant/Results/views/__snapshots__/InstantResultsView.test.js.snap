// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstantResultsView rendering specification InstantResultsView is rendered 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h3>
      File type not supported or it's not a banana, try again
    </h3>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        variant="outlined"
      >
        Back
      </button>
    </a>
  </header>
</div>
`;
